azext_iot/operations/_mqtt.py

```diff
@@ -4,95 +4,142 @@
+            if not prop.startswith("$"):
+                modified_properties[prop] = patch[prop]
+
+        if modified_properties:
+            print("\nTwin patch handler [Updating device twin reported properties]:")
+            self.printer.pprint(modified_properties)
+            self.device_client.patch_twin_reported_properties(modified_properties)
+
+    def execute(
+        self, data, properties={}, publish_delay=2, msg_count=100
+    ):
+        from tqdm import tqdm
 
-        sas = SasTokenAuthentication(
-            target["entity"],
-            target["policy"],
-            target["primarykey"],
-            sas_duration,
-        ).generate_sas_token()
-        cwd = EXTENSION_ROOT
-        cert_path = os.path.join(cwd, "digicert.pem")
-        tls = {"ca_certs": cert_path, "tls_version": ssl.PROTOCOL_SSLv23}
-        self.topic_publish = "devices/{}/messages/events/{}".format(
-            device_id, url_encode_dict(properties) if properties else ""
-        )
-        self.topic_receive = "devices/{}/messages/devicebound/#".format(device_id)
-        self.connected = False
-
-        self.client = mqtt.Client(protocol=mqtt.MQTTv311, client_id=device_id)
-        self.client.on_connect = self.on_connect
-        self.client.on_message = self.on_message
-        self.client.on_publish = self.on_publish
-        self.client.tls_set(ca_certs=tls["ca_certs"], tls_version=tls["tls_version"])
-        self.client.username_pw_set(username=build_mqtt_device_username(target["entity"], device_id), password=sas)
-        self.client.connect(host=self.target["entity"], port=8883)
-
-    def on_connect(self, client, userdata, flags, rc):
-        print(
-            "Connected to target IoT Hub with result: {}".format(
-                connection_result[rc]
-            )
-        )
-        self.client.subscribe(self.topic_receive)
-        print("Subscribed to device bound message queue")
-        self.connected = True
-
-    def on_message(self, client, userdata, msg):
-        print()
-        print("_Received C2D message with topic_: {}".format(msg.topic))
-        print("_Payload_: {}".format(msg.payload))
-
-    def on_publish(self, client, userdata, mid):
-        print(".", end="", flush=True)
-
-    def is_connected(self):
-        return self.connected
-
-    def execute(self, data, publish_delay=2, msg_count=100):
         try:
-            msgs = 0
-            self.client.loop_start()
-            while True:
-                if self.is_connected():
-                    if msgs < msg_count:
-                        msgs += 1
-                        self.client.publish(self.topic_publish, data.generate(True))
-                    else:
-                        break
+            if self.init_reported_properties:
+                self.device_client.patch_twin_reported_properties(self.init_reported_properties)
+
+            for _ in tqdm(range(0, msg_count), desc='Device simulation in progress', ascii=' #'):
```

Comment:
Why `ascii=' #'` over `ascii=True` as indicated in tqdm [known issues](https://github.com/tqdm/tqdm#faq-and-known-issues)?

Reply:
Doing ascii = True displays random numbers in the progress bar (which will be confusing for user). For instance we see a 6 after the hashes below and the number keeps changing randomly.

![image](https://user-images.githubusercontent.com/51140335/130508688-7bbcb0c2-30fa-43bf-8d11-39dff685df38.png)

I was able to fix this by using ``` ascii = ' #' ```